server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1372,6 +1372,68 @@ public static ByteSizeValue parseByteSize(String s, ByteSizeValue minValue, Byte
         return value;
     }
 
+    public static <T extends Enum<T>> Setting<T> enumSetting(Class<T> clazz, String key, T defaultValue, Property... properties) {
+        return new Setting<>(key, defaultValue.toString(), e -> parseEnum(clazz, key, e, isFiltered(properties)), properties);
+    }
+
+    public static <T extends Enum<T>> Setting<T> enumSetting(
+        Class<T> clazz,
+        String key,
+        Setting<T> fallbackSetting,
+        Property... properties
+    ) {
+        return new Setting<>(key, fallbackSetting, e -> parseEnum(clazz, key, e, isFiltered(properties)), properties);
+    }
+
+    public static <T extends Enum<T>> Setting<T> enumSetting(
+        Class<T> clazz,
+        String key,
+        T defaultValue,
+        Validator<T> validator,
+        Property... properties
+    ) {
+        return new Setting<>(key, defaultValue.toString(), e -> parseEnum(clazz, key, e, isFiltered(properties)), validator, properties);
+    }
+
+    public static <T extends Enum<T>> Setting<T> enumSetting(
+        Class<T> clazz,
+        String key,
+        Setting<T> fallbackSetting,
+        Validator<T> validator,
+        Property... properties
+    ) {
+        return new Setting<>(
+            new SimpleKey(key),
+            fallbackSetting,
+            fallbackSetting::getRaw,
+            e -> parseEnum(clazz, key, e, isFiltered(properties)),
+            validator,
+            properties
+        );
+    }
+
+    public static <T extends Enum<T>> Setting<T> enumSetting(
+        Class<T> clazz,
+        String key,
+        Function<Settings, String> defaultValueFn,
+        Property... properties
+    ) {
+        return new Setting<>(key, defaultValueFn, e -> parseEnum(clazz, key, e, isFiltered(properties)), properties);
+    }
+
+    static <T extends Enum<T>> T parseEnum(Class<T> clazz, String key, String value, boolean isFiltered) {
```

Comment:
I don't think the isFiltered should be necessary. The filtered property is for settings which are sensitive (it predates the ES keystore). I don't think we need that for enum settings.